astropy/modeling/bounding_box.py

```diff
@@ -9,13 +9,24 @@
 import abc
 import copy
 import warnings
-from typing import Any, Callable, NamedTuple
+from typing import TYPE_CHECKING, NamedTuple
 
 import numpy as np
 
 from astropy.units import Quantity
 from astropy.utils import isiterable
 
+if TYPE_CHECKING:
+    from typing import Any, Callable
+
+    from astropy.units import UnitBase
+    from astropy.utils.compat import PYTHON_LT_3_11
```

Comment:
For a separate discussion, but ruff has automatic pyupagrade handling of `sys.version_info`. Using our custom equivalents will require manual upgrades.

Reply:
[`pyupgrade` does indeed recognize `sys.version_info`](https://github.com/asottile/pyupgrade#python2-and-old-python3x-blocks), so it's better to use that.